File: algorithms/golang/two-sum.go
package main

import (
  "fmt"
)

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	res := twoSum(nums, target)
	fmt.Println(res)
}

func twoSum(nums []int, target int) []int {
	array := make(map[int]int)
	for key, num := range nums {
		search, ok := array[target - num]
		if ok {
			return []int {search, key}
		}
		array[num] = key
	}
	return nil
}

File: algorithms/php/two-sum.php
<?php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    static function twoSum($nums, $target) {
      $arr = [];
      foreach ($nums as $key => $num) {
        if (isset($arr[$target - $num])) {
          return [$arr[$target - $num], $key];
        }
        $arr[$num] = $key;
      }
    }
}

print_r(Solution::twoSum([2, 7, 11, 15], 9));
